File: sample/src/main/java/com/emarsys/sample/SampleCartItem.java
package com.emarsys.sample;

import com.emarsys.predict.api.model.CartItem;

public class SampleCartItem implements CartItem {
    private double price;
    private double quantity;
    private String itemId;

    SampleCartItem(String itemId, double price, double quantity) {
        this.itemId = itemId;
        this.price = price;
        this.quantity = quantity;
    }

    @Override
    public String getItemId() {
        return itemId;
    }

    @Override
    public double getPrice() {
        return price;
    }

    @Override
    public double getQuantity() {
        return quantity;
    }

    @Override
    public String toString() {
        return "CartItem{itemId:" + itemId + ", price:" + price + ", quantity:" + quantity + "}";
    }

}

File: sample/src/main/kotlin/com/emarsys/sample/predict/cart/SampleCartItem.kt
package com.emarsys.sample.predict.cart

import com.emarsys.predict.api.model.CartItem

data class SampleCartItem(
    override val itemId: String,
    override val price: Double,
    override val quantity: Double
) : CartItem {

    override fun toString(): String {
        return "Sample Cart Item: id = $itemId"
    }
}